android/sdk/src/main/java/com/taobao/weex/ui/component/WXComponent.java

```diff
@@ -1682,12 +1684,33 @@ public void setBackgroundImage(@NonNull String bgImage) {
       getOrCreateBorder().setImage(shader);
     }
   }
+  private boolean shouldCancelHardwareAccelerate() {
+    IWXConfigAdapter adapter = WXSDKManager.getInstance().getWxConfigAdapter();
+    boolean cancel_hardware_accelerate = false;
+    if (adapter != null) {
+      try {
+        cancel_hardware_accelerate = Boolean.parseBoolean(adapter
+                .getConfig("android_weex_test_gpu",
+                        "cancel_hardware_accelerate",
+                        "false"));
+      }catch (Exception e){
+        WXLogUtils.e(WXLogUtils.getStackTrace(e));
+      }
+      WXLogUtils.i("cancel_hardware_accelerate : " + cancel_hardware_accelerate);
+    }
+    return cancel_hardware_accelerate;
+  }
 
   public void setOpacity(float opacity) {
     if (opacity >= 0 && opacity <= 1 && mHost.getAlpha() != opacity) {
+      int limit = WXRenderManager.getOpenGLRenderLimitValue();
       if (isLayerTypeEnabled()) {
         mHost.setLayerType(View.LAYER_TYPE_HARDWARE, null);
       }
+      if(shouldCancelHardwareAccelerate() && limit > 0 && (getLayoutHeight() > limit ||
+              getLayoutWidth() > limit)){
+        mHost.setLayerType(View.LAYER_TYPE_SOFTWARE,null);
```

Comment:
I'm not sure which one you should choose `View.LAYER_TYPE_SOFTWARE` or `View.LAYER_TYPE_NONE`.